ethereum-network-partition-test/main_test.go

```diff
@@ -358,112 +462,122 @@ func waitUntilAllNodesGetSyncedBeforeInducingThePartition(
+	}
+	mostRecentBlockNumberForNode0, err := getMostRecentNodeBlockWithRetries(ctx, node0ServiceId, node0Client, retriesAttempts, retriesSleepDuration)
+	if err != nil {
+		return 0, stacktrace.NewError("Unable to retrieve current block for node '%s' after node diverged", node0ServiceId)
+	}
+	mostRecentBlockNumberForNodeN, err := getMostRecentNodeBlockWithRetries(ctx, nodeNServiceId, nodeNClient, retriesAttempts, retriesSleepDuration)
+	if err != nil {
+		return 0, stacktrace.NewError("Unable to retrieve current block for node '%s' after node diverged", nodeNServiceId)
+	}
+	if mostRecentBlockNumberForNode0.NumberU64() > mostRecentBlockNumberForNodeN.NumberU64() {
+		return mostRecentBlockNumberForNode0.NumberU64(), nil
+	} else {
+		return mostRecentBlockNumberForNodeN.NumberU64(), nil
 	}
-
-	return nil
 }
 
-func getNextNode0BlockNumberAndHash(
+func waitForNodesToProduceBlockNumberAfterPartitionWasIntroduced(
 	ctx context.Context,
+	node0ServiceId services.ServiceID,
 	node0Client *ethclient.Client,
+	nodeNServiceId services.ServiceID,
 	node2Client *ethclient.Client,
-	previousSyncedBlockNumber uint64,
-) (uint64, string, error) {
+	blockNumberToWaitForOnEachNode uint64,
+) (uint64, string, uint64, string, error) {
 
 	var wg sync.WaitGroup
 	ethNodeBlocksByServiceId := &sync.Map{}
-	errorChanForCheckingNode0 := make(chan error, 1)
-	defer close(errorChanForCheckingNode0)
+	errorChan := make(chan error, 1)
+	defer close(errorChan)
 
-	for true {
+	node0ReachedExpectedBlockNumber := false
+	nodeNReachedExpectedBlockNumber := false
+	for !node0ReachedExpectedBlockNumber || !nodeNReachedExpectedBlockNumber {
 		select {
+		// In Eth2, a new block is produced every 12 seconds. Ticking every second is more than enough
 		case <-time.Tick(1 * time.Second):
 			wg.Add(2)
-
 			//node0
 			go func() {
 				defer wg.Done()
-				block, err := getMostRecentNodeBlockWithRetries(ctx, node0Id, node0Client, retriesAttempts, retriesSleepDuration)
+				logrus.Debugf("Checking block number for node '%s'", node0ServiceId)
+				if node0ReachedExpectedBlockNumber {
+					logrus.Debugf("Node '%s' already reached block number '%d'", node0ServiceId, blockNumberToWaitForOnEachNode)
+					return
+				}
+				block, err := getMostRecentNodeBlockWithRetries(ctx, node0ServiceId, node0Client, retriesAttempts, retriesSleepDuration)
 				if err != nil {
-					errorChanForCheckingNode0 <- stacktrace.Propagate(err, "An error occurred getting the mos recent node bloc for service '%v'", node0Id)
+					errorChan <- stacktrace.Propagate(err, "An error occurred getting the mos recent node bloc for service '%v'", node0ServiceId)
```

Comment:
s/bloc/block

should this have a `return` statement here?

Reply:
Yup you're right, good catch